### PR TITLE
Chain Manager: remove casttypes for addresses in proto

### PR DESCRIPTION
**Pull request by murthyvitwit**

### CODE REVIEW

File: helper/call.go [hunk 1]
```diff
@@ -50,12 +49,12 @@ type IContractCaller interface {
 	DecodeNewHeaderBlockEvent(common.Address, *ethTypes.Receipt, uint64) (*rootchain.RootchainNewHeaderBlock, error)
 	// decode validator events
 	DecodeValidatorTopupFeesEvent(common.Address, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoTopUpFee, error)
-	DecodeValidatorJoinEvent(sdk.AccAddress, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoStaked, error)
-	DecodeValidatorStakeUpdateEvent(sdk.AccAddress, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoStakeUpdate, error)
-	DecodeValidatorExitEvent(sdk.AccAddress, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoUnstakeInit, error)
-	DecodeSignerUpdateEvent(sdk.AccAddress, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoSignerChange, error)
+	DecodeValidatorJoinEvent(common.Address, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoStaked, error)
```
Comment (anilCSE) [inline helper/call.go:52]: This should be `sdk.AccAddress`. Let's avoid `common.Address` usage

File: helper/call.go [hunk 2]
```diff
@@ -50,12 +49,12 @@ type IContractCaller interface {
...
+	DecodeValidatorStakeUpdateEvent(common.Address, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoStakeUpdate, error)
+	DecodeValidatorExitEvent(common.Address, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoUnstakeInit, error)
+	DecodeSignerUpdateEvent(common.Address, *ethTypes.Receipt, uint64) (*stakinginfo.StakinginfoSignerChange, error)
 	// decode state events
-	DecodeStateSyncedEvent(sdk.AccAddress, *ethTypes.Receipt, uint64) (*statesender.StatesenderStateSynced, error)
+	DecodeStateSyncedEvent(common.Address, *ethTypes.Receipt, uint64) (*statesender.StatesenderStateSynced, error)
```
Comment (anilCSE) [inline helper/call.go:57]: ditto

File: helper/call.go [hunk 3]
```diff
@@ -503,7 +502,7 @@ func (c *ContractCaller) DecodeValidatorTopupFeesEvent(contractAddress common.Ad
 }
 
 // DecodeValidatorJoinEvent represents validator staked event
-func (c *ContractCaller) DecodeValidatorJoinEvent(contractAddress sdk.AccAddress, receipt *ethTypes.Receipt, logIndex uint64) (*stakinginfo.StakinginfoStaked, error) {
+func (c *ContractCaller) DecodeValidatorJoinEvent(contractAddress common.Address, receipt *ethTypes.Receipt, logIndex uint64) (*stakinginfo.StakinginfoStaked, error) {
```
Comment (anilCSE) [inline helper/call.go:505]: ditto

File: proto/chainmanager/v1beta1/genesis.proto [hunk 4]
```diff
@@ -11,38 +11,21 @@ option (gogoproto.unmarshaler_all) = true;
...
-    string state_sender_address = 7
-        [(gogoproto.casttype) =
-             "github.com/cosmos/cosmos-sdk/types.AccAddress"];
-    string state_receiver_address = 8
-        [(gogoproto.casttype) =
-             "github.com/cosmos/cosmos-sdk/types.AccAddress"];
-    string validator_set_address = 9
-        [(gogoproto.casttype) =
-             "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    string matic_token_address = 2;
```
Comment (anilCSE) [inline proto/chainmanager/v1beta1/genesis.proto:21]: :+1:

File: proto/chainmanager/v1beta1/genesis.proto [hunk 5]
```diff
@@ -51,5 +34,5 @@ message Params {
 
     uint64      mainchain_tx_confirmations  = 1;
     uint64      maticchain_tx_confirmations = 2;
-    ChainParams chain_params                = 3;
+    ChainParams chain_params                = 3 [(gogoproto.nullable) = false];
```
Comment (anilCSE) [inline proto/chainmanager/v1beta1/genesis.proto:37]: :+1:

File: x/staking/side_handler.go [hunk 6]
```diff
@@ -86,7 +86,9 @@ func SideHandleMsgValidatorJoin(ctx sdk.Context, msg types.MsgValidatorJoin, k k
 	}
 
 	// decode validator join event
-	eventLog, err := contractCaller.DecodeValidatorJoinEvent(chainParams.StakingInfoAddress, receipt, msg.LogIndex)
+	var stakingInfoAddress [20]byte
+	copy(stakingInfoAddress[:], chainParams.StakingInfoAddress)
+	eventLog, err := contractCaller.DecodeValidatorJoinEvent(stakingInfoAddress, receipt, msg.LogIndex)
```
Comment (anilCSE) [inline x/staking/side_handler.go:91]: this looks hacky, any better way to handle this?

**Review by anilCSE — CHANGES_REQUESTED** [review]

File: x/staking/side_handler.go (hunk 6 above)
Comment (anilCSE) [reply x/staking/side_handler.go:91]: @jdkanani

File: helper/call.go (hunk 1 above)
Comment (anilCSE) [reply helper/call.go:52]: @jdkanani is there a reason to maintain `common.Address` here?

Comment (jdkanani) [reply helper/call.go:52]: `common.Address` is from Bor (Geth lib). It is required for contract go files, which are generated from Geth directly. ABI to go files. Unfortunately, we won't be able to change it.

File: x/staking/side_handler.go (hunk 6 above)
Comment (jdkanani) [reply x/staking/side_handler.go:91]: We have to decode events and validator them in the side handler. Not sure how to get rid of it.

Comment (anilCSE) [reply x/staking/side_handler.go:91]: @murthyvitwit for now let's create a util function and use it from there

File: helper/call.go (hunk 1 above)
Comment (jdkanani) [reply helper/call.go:52]: Ignore my previous comment @anilCSE. We can use `sdk.AccAddress` here.

File: helper/call.go (hunk 2 above)
Comment (jdkanani) [reply helper/call.go:57]: Agree.

File: helper/call.go (hunk 3 above)
Comment (jdkanani) [reply helper/call.go:504]: Agree.

File: x/staking/side_handler.go (hunk 6 above)
Comment (jdkanani) [reply x/staking/side_handler.go:91]: Sorry for the misunderstanding. We can avoid conversion.

File: helper/call.go (hunk 1 above)
Comment (murthyvitwit) [reply helper/call.go]: I changed to use **sdk.AccAddress**

File: x/staking/side_handler.go (hunk 6 above)
Comment (murthyvitwit) [reply x/staking/side_handler.go]: removed it

**Review by jdkanani — APPROVED** [review]

**Review by anilCSE — APPROVED** [review]

**Pull request merged**